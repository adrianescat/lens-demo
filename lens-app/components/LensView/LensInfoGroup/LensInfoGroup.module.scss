.contentGroup {
  margin-bottom: 30px;
}

.contentGroupTitle {
  display: table;
  padding-left: 40px;
  text-transform: uppercase;
  font-family: 'MSemiBold';
  color: #a9a9a9;
  margin-bottom: 0.5rem;
  font-size: 10px;
  letter-spacing: 1px;
}

.contentGroupItem {
  position: relative;

  .delete {
    display: none;
    opacity: 0;
    width: 0;
  }

  .trashIcon {
    width: 12px;
  }

  .actionMenu {
    display: none;
  }

  .actionLabel {
    float: rigth;
    opacity: 0;
    visibility: none;
    text-transform: uppercase;
    font-size: 9.75px;
    color: #a9a9a9;
    font-family: 'MSemiBold';
    letter-spacing: 1px;
  }
}

.contentGroupItem.active {
  box-shadow: 0px 10px 17px 0px #f3f3f3;
  width: 90%;
  max-height: 15rem;
  z-index: 2;
  margin: 0 auto;

  .actionMenu {
    display: block;
    float: right;
  }

  .groupItemSubContent,
  .groupItemSubContentDone {
    display: flex;
    justify-content: space-between;
    align-items: center;
    flex-direction: row;
  }

  .groupItemSubContentDone {
    justify-content: flex-end;
  }

  .groupItemName {
    border-bottom-color: #f0f0f0;
  }

  &:hover {
    .delete {
      display: inline-block;
      animation: moveIt .3s cubic-bezier(0.55, 0.6, 0.34, 0.95) forwards;
    }

    .actionLabel {
      opacity: 1;
      visibility: visible;
      transition: opacity 2s cubic-bezier(0.15, 0.9, 0.34, 0.95);
    }

    .groupItemName:hover {
      background-color: #fff;
    } 
  }
}

.groupItemName {
  cursor: pointer;
  display: table;
  width: 100%;
  font-size: 14px;
  padding-left: 40px;
  padding-right: 40px;
  padding-top: 14px;
  border-bottom: 1px solid transparent;
  padding-bottom: 14px;
  font-family: 'MMedium';
  color: #313131;
  position: relative;

  &:hover {
    background-color: #f6f6f6;
    border-color: #f6f6f6;
  }
}

.groupItemSubContent,
.groupItemSubContentDone {
  display: none;
  height: 58px;
  padding: 0 30px;

  .subContentCancel,
  .subContentConfirm,
  .subContentRemove {
    font-family: 'MSemiBold';
    font-size: 12px;
    letter-spacing: 1px;
    text-transform: uppercase;
    color: #303030;
    cursor: pointer;
  }
}

@keyframes moveIt {
  0% {
    width: 0;
  }
  20% {
    width: 5px;
  }
  40% {
    width: 10px;
  }
  60% {
    width: 13px;
  }
  70% {
    width: 15px;
  }
  75% {
    width: 16px;
  }
  80% {
    width: 17px;
  }
  100% {
    width: 20px;
    opacity: 1;
  }
}