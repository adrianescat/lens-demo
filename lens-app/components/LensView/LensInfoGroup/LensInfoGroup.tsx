import React, { Dispatch, ReactElement, SetStateAction } from 'react'
import classnames from 'classnames';
import { faTrashAlt } from '@fortawesome/free-solid-svg-icons'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import styles from './LensInfoGroup.module.scss';
import {
  Lens,
  File,
  Flag,
  Problem,
  Allergy,
  Medication,
  flagActions,
} from '../../../types/lens';
import PopoverMenu from '../../PopoverMenu';

interface Props {
  groupName: string;
  list: Array<any>;
  listDone: Array<Flag | Problem | Allergy | Medication>;
  activeDot: string,
  actions?: Array<any>
  hasActions?: boolean;
  isDoneList?: boolean;
  listSetter: Dispatch<SetStateAction<[] | Flag[]>>;
  handleClearDot: () => void;
  listDoneSetter: Dispatch<SetStateAction<[] | Flag[]>>;
  handleDotClick: (name: string) => void;
  handleRemoveItem: (name: string, collection, setter) => void;
  handleMoveItem: (name: string, collection, setter, collectionMoveTo, setterMoveTo) => void;
}

export default function LensInfoGroup({
  groupName,
  list,
  listDone,
  hasActions,
  isDoneList,
  actions,
  activeDot,
  listSetter,
  handleClearDot,
  listDoneSetter,
  handleDotClick,
  handleRemoveItem,
  handleMoveItem,
}: Props): ReactElement {
  if (list.length === 0) {
    return null;
  }
  console.log(groupName)
  return (
    <div className={styles.contentGroup}>
      <div className={styles.contentGroupTitle}>
        {groupName}
      </div>
      {list.map((item) => (
        <div
          key={item.name}
          className={classnames(styles.contentGroupItem, activeDot === item.name ? styles.active : '')}
        >
          <span
            className={styles.groupItemName}
            onClick={() => handleDotClick(item.name)}
          >
            {!isDoneList && <span
              className={styles.delete}
              onClick={() => handleRemoveItem(item.name, list, listSetter)}
            >
              <FontAwesomeIcon className={styles.trashIcon} icon={faTrashAlt} />
            </span>}
            {item.humanName}
            {hasActions && (item.moreInfo || actions) && <div className={styles.actionMenu}>
              <PopoverMenu
                list={item.moreInfo ? item.moreInfo : actions}
              >
                <span className={styles.actionLabel}>{item.moreInfo ? 'More Info' : 'Action'}</span>
              </PopoverMenu>
            </div>}
          </span>
          <div className={styles.groupItemSubContent}>
            <div
              className={styles.subContentCancel}
              onClick={handleClearDot}
            >
              Cancel
            </div>
            <div
              className={styles.subContentConfirm}
              onClick={() => handleMoveItem(item.name, list, listSetter, listDone, listDoneSetter)}
            >
              {isDoneList ? 'Remove' : 'Confirm'}
            </div>
          </div>
        </div>
      ))}
    </div>
  )
}
