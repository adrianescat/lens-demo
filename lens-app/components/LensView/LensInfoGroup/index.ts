import LensInfoGroup from './LensInfoGroup';

export default LensInfoGroup;