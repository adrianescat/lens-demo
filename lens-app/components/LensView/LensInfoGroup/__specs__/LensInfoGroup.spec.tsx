import React from 'react';
import { shallow } from 'enzyme';
import LensInfoGroup, { Props } from '../LensInfoGroup';

let wrapper: any;
let props: Props;

beforeEach(() => {
  props = {
    groupName: 'groupName',
    list: [
      {
        "name": "PrimaryCareProvider/FollowUp",
        "humanName": "Follow up",
        "confidence": 1.0,
        "weight": 0.7,
        "page": 0,
        "box": {
          "t": 40,
          "b": 720,
          "l": 90,
          "r": 1100
        }
      },
      {
        "name": "CardiacFaliure/Symptoms",
        "humanName": "Heart failure",
        "confidence": 1.0,
        "weight": 1.2,
        "page": 0,
        "box": {
          "t": 80,
          "b": 1320,
          "l": 20,
          "r": 635
        }
      }
    ],
    listDone: [],
    hasActions: true,
    isDoneList: false,
    actions: ['email', 'calendar'],
    activeDot: 'activeDot',
    listSetter: jest.fn(),
    handleClearDot: jest.fn(),
    listDoneSetter: jest.fn(),
    handleDotClick: jest.fn(),
    handleRemoveItem: jest.fn(),
    handleMoveItem: jest.fn(),
  }
  wrapper = shallow(<LensInfoGroup {...props} />);
});

test('should match snapshots', () => {
  expect(wrapper).toMatchSnapshot();
});