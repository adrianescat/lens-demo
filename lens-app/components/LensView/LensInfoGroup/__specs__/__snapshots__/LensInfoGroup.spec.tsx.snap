// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshots 1`] = `
<div
  className="contentGroup"
>
  <div
    className="contentGroupTitle"
  >
    groupName
  </div>
  <div
    className="contentGroupItem"
    key="PrimaryCareProvider/FollowUp"
  >
    <span
      className="groupItemName"
      onClick={[Function]}
    >
      <span
        className="delete"
        onClick={[Function]}
      >
        <FontAwesomeIcon
          border={false}
          className="trashIcon"
          fixedWidth={false}
          flip={null}
          icon={
            Object {
              "icon": Array [
                448,
                512,
                Array [],
                "f2ed",
                "M32 464a48 48 0 0 0 48 48h288a48 48 0 0 0 48-48V128H32zm272-256a16 16 0 0 1 32 0v224a16 16 0 0 1-32 0zm-96 0a16 16 0 0 1 32 0v224a16 16 0 0 1-32 0zm-96 0a16 16 0 0 1 32 0v224a16 16 0 0 1-32 0zM432 32H312l-9.4-18.7A24 24 0 0 0 281.1 0H166.8a23.72 23.72 0 0 0-21.4 13.3L136 32H16A16 16 0 0 0 0 48v32a16 16 0 0 0 16 16h416a16 16 0 0 0 16-16V48a16 16 0 0 0-16-16z",
              ],
              "iconName": "trash-alt",
              "prefix": "fas",
            }
          }
          inverse={false}
          listItem={false}
          mask={null}
          pull={null}
          pulse={false}
          rotation={null}
          size={null}
          spin={false}
          swapOpacity={false}
          symbol={false}
          title=""
          transform={null}
        />
      </span>
      Follow up
      <div
        className="actionMenu"
      >
        <PopoverMenu
          list={
            Array [
              "email",
              "calendar",
            ]
          }
        >
          <span
            className="actionLabel"
          >
            Action
          </span>
        </PopoverMenu>
      </div>
    </span>
    <div
      className="groupItemSubContent"
    >
      <div
        className="subContentCancel"
        onClick={[MockFunction]}
      >
        Cancel
      </div>
      <div
        className="subContentConfirm"
        onClick={[Function]}
      >
        Confirm
      </div>
    </div>
  </div>
  <div
    className="contentGroupItem"
    key="CardiacFaliure/Symptoms"
  >
    <span
      className="groupItemName"
      onClick={[Function]}
    >
      <span
        className="delete"
        onClick={[Function]}
      >
        <FontAwesomeIcon
          border={false}
          className="trashIcon"
          fixedWidth={false}
          flip={null}
          icon={
            Object {
              "icon": Array [
                448,
                512,
                Array [],
                "f2ed",
                "M32 464a48 48 0 0 0 48 48h288a48 48 0 0 0 48-48V128H32zm272-256a16 16 0 0 1 32 0v224a16 16 0 0 1-32 0zm-96 0a16 16 0 0 1 32 0v224a16 16 0 0 1-32 0zm-96 0a16 16 0 0 1 32 0v224a16 16 0 0 1-32 0zM432 32H312l-9.4-18.7A24 24 0 0 0 281.1 0H166.8a23.72 23.72 0 0 0-21.4 13.3L136 32H16A16 16 0 0 0 0 48v32a16 16 0 0 0 16 16h416a16 16 0 0 0 16-16V48a16 16 0 0 0-16-16z",
              ],
              "iconName": "trash-alt",
              "prefix": "fas",
            }
          }
          inverse={false}
          listItem={false}
          mask={null}
          pull={null}
          pulse={false}
          rotation={null}
          size={null}
          spin={false}
          swapOpacity={false}
          symbol={false}
          title=""
          transform={null}
        />
      </span>
      Heart failure
      <div
        className="actionMenu"
      >
        <PopoverMenu
          list={
            Array [
              "email",
              "calendar",
            ]
          }
        >
          <span
            className="actionLabel"
          >
            Action
          </span>
        </PopoverMenu>
      </div>
    </span>
    <div
      className="groupItemSubContent"
    >
      <div
        className="subContentCancel"
        onClick={[MockFunction]}
      >
        Cancel
      </div>
      <div
        className="subContentConfirm"
        onClick={[Function]}
      >
        Confirm
      </div>
    </div>
  </div>
</div>
`;
